src/main/java/util/TickingTimerManager.java

```diff
@@ -0,0 +1,65 @@
+package util;
+
+import ui.UI;
+
+/**
+ * This class is used to hold the TickingTimer instance and perform operations associated with it.
+ */
+public class TickingTimerManager {
+
+    public static final int BUFFER_TIME = 1; //To allow some additional time until next refresh.
+    private final TickingTimer timer;
+    private static final int REFRESH_PERIOD = 60;
+
+    public TickingTimerManager(TickingTimer timer){
+        this.timer = timer;
+    }
+
+    public void startTimer(){
+        timer.start();
+    }
+
+    public void restartTimer(){
+        timer.restart();
+    }
+
+
+    /* This function updates the TickerTimer period, which is used to periodically refresh the issues.
+        * @param apiQuota : The remaining allowed api request until the next api request allowance renewal.
+        * @param remainingTime : The remaining time left until the next api request allowance renewal.
+        * @param lastApiCallsUsed : The amount of api used in the last api pull.
+        */
+   public int updateTickerTimer(int apiQuota, long remainingTime, int lastApiCallsUsed){
+
+       long refreshTimeInMins;
+
+       if (lastApiCallsUsed > 0) {
+           // If less or equal than APIQUOTA_BUFFER, restricts auto refresh until next apiQuota renewal.
+           // The APIQUOTA_BUFFER is reserved for Manual refresh and creation of issues, etc.
```

Comment:
These behaviour should be commented where the variables `APIQUOTA_BUFFER` AND `BUFFER_TIME` is declared instead of used to describe the control flow here.